test/vm/README.md

```diff
@@ -0,0 +1,349 @@
+
+# VM - Driver Functionality Test Suites - Kernel Compatibility Tests
+
+
+![Architectures](https://img.shields.io/badge/ARCHS-x86__64-blueviolet?style=for-the-badge)
+
+Status: **Under development, experimental**
+
+
+## Running VM-based Driver Functionality Tests on `localhost`
+
+### System Requirements (Host OS)
+
+It is strongly advised to follow the installation instructions in the official documentation and to run official smoke-tests for every dependency. See [Support Matrix](#support-matrix) for supported host and guest OS / VM specifications.
+
+- [Docker](https://www.docker.com/products/docker-desktop/) >= 20.10.9
+	- Configure docker to run as non-root user
+- [VirtualBox](https://www.virtualbox.org/wiki/Downloads) VM provider, latest version
+- [Vagrant](https://www.vagrantup.com/), latest version
+	- [vagrant virtualbox](https://www.vagrantup.com/docs/providers/virtualbox)
+	- `vagrant plugin install vagrant-vbguest`
+- [Go](https://go.dev/) >= 1.15
+- Ensure that necessary binaries are available on your host machine, such as, `bash`, `time`, `ssh`, `scp`, `tar`.
+
+### CMake Targets
+
+The test suites are integrated with the project's CMake setup but function independently. The primary VM build output is stored in the `libs/test/vm/build` directory to cache results, even if the `libs/build` folder is regenerated. The builds for scap-open and the kernel drivers are performed from scratch in containers, not using the `libs/build` directory. Further information about each step is available in later sections of this document.
+
+As a rule of thumb, test flakiness mostly arises due to failed kernel module (kmod) tests. These tests are best run on a powerful Linux box.
+
+
+```bash
+mkdir -p build;
+cd build;
+cmake -DCREATE_TEST_TARGETS=ON -DENABLE_VM_TESTS=ON ../;
+```
+
+Create containers, download kernel and header packages, extract kernel headers, and build vagrant VMs.
+
+```bash
+# Check dependencies
+make vm-dependency-check;
```

Comment:
Hey @incertum, after running the commands above I don't find this target. Should it require some steps furter?